fe1-web/src/core/components/QrCodeScanner.tsx

```diff
@@ -54,53 +49,85 @@ const styles = StyleSheet.create({
   },
 });
 
-// TODO: Fix react-qr-reader
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
 const QrCodeScanner = ({ showCamera, children, handleScan }: IPropTypes) => {
-  // const toast = useToast();
-  const [facingMode, setFacingMode] = useState<'user' | 'environment'>('user');
-  const [hasMultipleCameras, setHasMultipleCameras] = useState(false);
+  const [permission, requestPermission] = Camera.useCameraPermissions();
+  const [cameraType, setCameraType] = useState<CameraType>(CameraType.back);
+  const [hasMultipleCameras, setHasMultipleCameras] = useState(true);
+  const [hasCamera, setHasCamera] = useState(true);
 
   useEffect(() => {
-    try {
-      getNavigator()
-        .mediaDevices.enumerateDevices()
-        .then((devices) => {
-          console.log(devices.filter((device) => device.kind === 'videoinput'));
-          if (devices.filter((device) => device.kind === 'videoinput').length > 1) {
-            setHasMultipleCameras(true);
-          }
-        })
-        .catch(console.error);
-    } catch (e) {
-      // the browser might not support this api
-    }
+    (async () => {
+      if (permission && !permission.granted) {
+        await requestPermission();
+      }
+    })();
+  }, [permission, requestPermission]);
+
+  useEffect(() => {
+    let isMounted = true;
+    (async () => {
+      const isAvailable = await Camera.isAvailableAsync();
+      if (isMounted) {
+        setHasCamera(isAvailable);
+      } else {
+        return;
+      }
+      if (isAvailable) {
+        const types = await Camera.getAvailableCameraTypesAsync();
+        if (isMounted) setHasMultipleCameras(types.length > 1);
+      }
+    })();
+    return () => {
+      isMounted = false;
+    };
   }, []);
 
-  /*
-  const handleError = (err: string | Error) => {
-    console.error(err);
-    toast.show(err.toString(), {
-      type: 'danger',
-      placement: 'top',
-      duration: FOUR_SECONDS,
-    });
+  if (!hasCamera) {
+    return (
+      <>
+        <Text>Camera unavailable</Text>
+        <View style={styles.children}>{children}</View>
+      </>
+    );
+  }
+
+  if (!permission) {
+    return (
+      <>
+        <Text>Requesting for camera permissions</Text>
+        <View style={styles.children}>{children}</View>
+      </>
+    );
+  }
+
+  if (!permission.granted) {
+    return (
+      <>
+        <Text>Permissions for camera denied</Text>
+        <View style={styles.children}>{children}</View>
+      </>
+    );
+  }
+
+  // Scan each code only once
+  let lastValue: string;
+  const onBarCodeScanned = (result: BarCodeScanningResult) => {
+    if (lastValue && lastValue === result.data) return;
+    handleScan(result);
```

Comment:
as we had discussed, it makes sense here to just return `result.data` , so that the other APIs don't need to know about `BarCodeScanningResult`, which is an implementation detail from their perspective.